data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -473,18 +475,20 @@ private SerializedJson getDocument(final Event event) {
       }
     }
 
-    String routing = null;
+    String routingValue = null;
     if (routingField != null) {
-        if (expressionEvaluator.isValidFormatExpression(routingField)) {
-            routing = event.formatString(routingField, expressionEvaluator);
+        routingValue = event.get(routingField, String.class);
+    } else if (routing != null) {
+        if (expressionEvaluator.isValidFormatExpression(routing)) {
```

Comment:
I think this function will return true for static values such as `routing: "my_value"`. Should we change this function to be an invalid expression when there is no `${}` at all?